_plugins/eol.rb

```diff
@@ -0,0 +1,106 @@
+require 'jekyll'
+require 'date'
+require 'pp'
+
+# This plugin enriches the product pages
+# by parsing the YAML and setting some standard
+# fields that can be easily consumed by the API
+# and other layouts
+module Jekyll
+  class Product
+    class << self
+      def enrich(page)
+        if page['iconSlug']
+          page.data['iconUrl'] = "https://simpleicons.org/icons/#{page['iconSlug']}.svg"
+        end
+
+        set_defaults(page)
+        set_column_headers(page)
+        enrich_releases(page)
+      end
+
+      def set_defaults(page)
+        page.data['shortlink'] ||= find_shortlink(page)
+      end
+
+      def find_shortlink(page)
+        return page['permalink'] unless page['alternate_urls']
+        shortlink = page['permalink']
+        page['alternate_urls'].each do |u|
+          shortlink = u if u.size < shortlink.size
+        end
+        shortlink
+      end
+
+      def enrich_releases(page)
+        today = Date.today
+        page['releases'].each do |r|
+          r['text'] = {}
+          r['colors'] = {}
+          ['eol', 'support'].each do |x|
```

Comment:
I am not fan of computing colors in the plugin: colors must also be computed for values other than `eol` and `support` such as `discontinued` or custom columns.

IMHO it will be more reusable if it's implemented in a filter such as in #2229.